GitUI/Avatars/FileSystemAvatarCache.cs

```diff
@@ -13,11 +13,26 @@ public sealed class FileSystemAvatarCache : IAvatarProvider, IAvatarCacheCleaner
 
         private readonly IAvatarProvider _inner;
         private readonly IFileSystem _fileSystem;
+        private readonly string _cacheDir;
+        private readonly DateTime _expirationDate;
 
         public FileSystemAvatarCache(IAvatarProvider inner, IFileSystem? fileSystem = null)
         {
             _inner = inner;
             _fileSystem = fileSystem ?? new FileSystem();
+            int avatarImageCacheDays = AppSettings.AvatarImageCacheDays;
+            if (avatarImageCacheDays < 1)
+            {
+                avatarImageCacheDays = DefaultCacheDays;
+            }
+
+            _expirationDate = DateTime.Now.AddDays(-avatarImageCacheDays);
```

Comment:
This is counter-intuitive... An expiration date to me is a date in a future, whereas here it's a date in a past.

Reply:
I don't really see why. 
Is it a problem of naming and you have a better name than "expiration"? "obsolete"? "out of date"?
For me it's perfectly fine to compare if the file last creation time is older than a date in the past.

Reply:
An expiry date is a typically date in a future after which something shouldn't be used. In this instance we have a date in a past...
I'm struggling to understand the meaning of this field.

Reply:
We are trying to check if a file on the filesystem is old. I find it much easier to understand and logical to compare the modification date to check if the date is older than a date in the past than adding days to this date and check if it is a date in the future. And I should not be the only one because that's how it has been written initially as I have not changed this logic...

This field is the date before which we consider the avatar too old and that we compare to the last write time of the avatar file. This (initial) logic seems good to me...